AK/JsonObject.h

```diff
@@ -100,6 +100,9 @@ class JsonObject {
 
     void set(const String& key, JsonValue value)
     {
+        auto key_index = m_order.find_first_index(key);
+        if (key_index.has_value())
+            m_order.remove(key_index.value());
```

Comment:
Is it correct behavior that the order changes when you replace an entry?
Also, you can use the return value from `m_members.set` to know if you've just replaced an existing entry :)

Reply:
> Is it correct behavior that the order changes when you replace an entry?

Debatable, but that's what I'd expect.

```cpp
json.set("some", "foo");
json.set("key", "bar");
json.set("some", "baz");
```

should IMO serialize to

```json
{"key": "bar", "some": "baz"}
```

```json
{"some": "foo", "key": "bar", "some": "baz"}
```

OR

but not

```json
{"some": "baz", "key": "bar"}
```

>  Also, you can use the return value from m_members.set to know if you've just replaced an existing entry :)

Neat! will update :+1: